Mathlib/LinearAlgebra/Matrix/Charpoly/Basic.lean

```diff
@@ -112,6 +118,19 @@ lemma charpoly_fromBlocks_zero₂₁ :
   simp only [charpoly, charmatrix_fromBlocks, Matrix.map_zero _ (Polynomial.C_0), neg_zero,
     det_fromBlocks_zero₂₁]
 
+lemma toSquareBlock_charmatrix {α : Type*} [DecidableEq α] {b : n → α} {a : α}  :
+    (M.toSquareBlock b a).charmatrix = M.charmatrix.toSquareBlock b a := by
+  ext i j : 1
+  simp [charmatrix_apply, toSquareBlock_def, diagonal_apply, Subtype.ext_iff]
+
+lemma BlockTriangular.charpoly {α : Type*} {b : n → α} [LinearOrder α] (h : M.BlockTriangular b) :
+    M.charpoly = ∏ a ∈ image b univ, (M.toSquareBlock b a).charpoly := by
```

Comment:
Is `image _ univ` not usually spelled `range`?

Reply:
`Set.image _ Set.univ` is, but not `Finset.image _ Finset.univ`.